src/xtgeo/grid3d/_grid_import.py

```diff
@@ -1,54 +1,77 @@
 """Grid import functions for various formats."""
+from __future__ import annotations
+
+from pathlib import Path
+from typing import Any, Literal
 
-import xtgeo
 from xtgeo.common import null_logger
+from xtgeo.common.sys import _XTGeoFile
 from xtgeo.grid3d import _grid_import_ecl, _grid_import_roff
 
 from . import _grid_import_xtgcpgeom
 
 logger = null_logger(__name__)
 
 
-def from_file(gfile, fformat=None, **kwargs):
+def from_file(
+    gfile: _XTGeoFile,
+    fformat: Literal[
+        "bgrdecl",
+        "egrid",
+        "fegrid",
+        "grdecl",
+        "guess",
+        "hdf",
+        "roff_ascii",
+        "roff_binary",
+        "xtg",
```

Comment:
Might move this a common types or something, but will keep them here for now. Needs abit more research before i do so.